hoomd/mpcd/SorterGPU.cu

```diff
@@ -168,97 +158,36 @@ cudaError_t sort_apply(Scalar4* d_pos_alt,
     }
 
 /*!
- * \param d_cell_list Cell list to fill in
- * \param d_cell_np Number of particles per cell
- * \param cli Two-dimensional cell-list indexer
- * \param sentinel Value to fill into empty cell-list entries
- * \param N_cli Number of total entries (filled and empty) in the cell list
+ * \param d_order Compacted MPCD particle indexes in cell-list order (output)
+ * \param d_cell_id auxillary array to sort cell id of each mpcd particles
+ * \param d_vel Cell indexes of the MPCD particles
+ * \param N Number of particles
  * \param block_size Number of threads per block
- *
+
  * \returns cudaSuccess on completion
  *
- * \sa mpcd::gpu::kernel::sort_set_sentinel
+ * \sa mpcd::gpu::kernel::set_order
  */
-cudaError_t sort_set_sentinel(unsigned int* d_cell_list,
-                              const unsigned int* d_cell_np,
-                              const Index2D& cli,
-                              const unsigned int sentinel,
-                              const unsigned int block_size)
+cudaError_t compute_order(unsigned int* d_order,
+                          unsigned int* d_cell_id,
+                          const Scalar4* d_vel,
+                          const unsigned int N,
+                          const unsigned int block_size)
     {
     unsigned int max_block_size;
     cudaFuncAttributes attr;
-    cudaFuncGetAttributes(&attr, (const void*)mpcd::gpu::kernel::sort_set_sentinel);
+    cudaFuncGetAttributes(&attr, (const void*)mpcd::gpu::kernel::set_order);
     max_block_size = attr.maxThreadsPerBlock;
 
-    const unsigned int N_cli = cli.getNumElements();
-
     unsigned int run_block_size = min(block_size, max_block_size);
-    dim3 grid(N_cli / run_block_size + 1);
-    mpcd::gpu::kernel::sort_set_sentinel<<<grid, run_block_size>>>(d_cell_list,
-                                                                   d_cell_np,
-                                                                   cli,
-                                                                   sentinel,
-                                                                   N_cli);
-
+    dim3 grid(N / run_block_size + 1);
+    mpcd::gpu::kernel::set_order<<<grid, run_block_size>>>(d_order, d_cell_id, d_vel, N);
+    thrust::device_ptr<unsigned int> order(d_order);
+    thrust::device_ptr<unsigned int> cell_id(d_cell_id);
+    thrust::sort_by_key(thrust::device, cell_id, cell_id + N, order);
```

Comment:
Your tuner won't be measuring quite the right thing if the calls to thrust for sorting are in the same function as the kernel you're timing. To be safe, I would put the thrust calls that do the sorting in a separate function than the one that fills in the initial order.